src/runapi.js

```diff
@@ -0,0 +1,61 @@
+const apiCommands = require("./tdx-api-commands");
+
+function objectToListMapper(obj) {
+  return Object.keys(obj).sort().map((element) => ([element, Number(element)]));
+}
+
+function stringifyArgumentValue(args, keyList) {
+  if (keyList.length > 1) {
+    const key = keyList.shift();
+    if (key in args) stringifyArgumentValue(args[key], keyList);
+  } else if (keyList.length === 1) {
+    const key = keyList.shift();
+    if (key in args) {
+      args[key] = (typeof args[key] === "number") ?
+        args[key].toString() :
+        args[key];
+    }
+  }
+}
+
+function stringifyArguments(apiArgs, apiArgsStringify) {
+  apiArgsStringify.forEach((element) => stringifyArgumentValue(apiArgs, element.split(".")));
+}
+
+function getFunctionArguments(args) {
+  const mapper = objectToListMapper(args);
+  let fullArgs = [];
+
+  mapper.forEach((element) => {
+    const argument = args[element[0]];
+    const argumentIdx = element[1];
```

Comment:
I'm fairly certain this is more legible. `for of` is faster than `Array~forEach`, so prefer instead of using an in-line function. And you can do `[list, destructuring]` as well [magick](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment).
 
```suggestion
  for (const [argumentKey, argumentIdx] of mapper) {
    const argument = args[argumentKey];
```

Reply:
This is true magick.